src/lib/hooks/useResize.js

```diff
@@ -0,0 +1,19 @@
+import { useEffect, useState } from "react";
+
+const useResize = ({ getPageSize }) => {
+  const [size, setSize] = useState(getPageSize(window.innerWidth));
+
+  const handleResize = () => {
+    const newWindowSize = getPageSize(window.innerWidth);
+    setSize(newWindowSize);
+  };
+
+  useEffect(() => {
+    window.addEventListener("resize", handleResize);
+    handleResize();
+    return () => window.removeEventListener("resize", handleResize);
+  }, []);
+  return size;
+};
```

Comment:
커스텀 훅으로 잘 만드셨네요 ! 앞으로 커스텀 훅 많이 많이 써보세요 ㅎㅎ 
리턴값은 객체로 리턴하는게 편합니다 ! 

```suggestion
const useResize = ({ getPageSize }) => {
  const [size, setSize] = useState(getPageSize(window.innerWidth));

  const handleResize = () => {
    const newWindowSize = getPageSize(window.innerWidth);
    setSize(newWindowSize);
  };

  useEffect(() => {
    window.addEventListener("resize", handleResize);
    handleResize();
    return () => window.removeEventListener("resize", handleResize);
  }, []);
  return { size }
};
```